### PR TITLE
Updates to symfony 4.0

### PR DESCRIPTION
**Pull request by JulianStier**
Migration preparations for symfony 4.0

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
 /vendor/
 /build
 composer.lock
+
+# IDE-specific
```
Comment (phansys) [inline .gitignore]: Why is this change required? Please, add environment specific files to your [global ignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) file instead.

File: composer.json [hunk 2]
```diff
@@ -12,18 +12,18 @@
     }
   ],
   "require": {
-    "php": ">=5.5.9",
+    "php": "^7.1",
     "monolog/monolog": "^1.0",
-    "symfony/console": "^3.0",
```
Comment (phansys) [inline composer.json]: Supporting version 4 of Symfony should not imply to drop support for version 3, since the features provided by version 3 are still compatible with the current codebase. Please, update the constraints to support both versions instead (`^3.0|^4.0`).

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,5 @@
 language: php
 php:
-  - '5.5'
```
Comment (phansys) [inline .travis.yml]: Support for these versions should not be dropped.

Comment (unherz) [reply .travis.yml]: Its right to drop support for php versions <= 7.0 because they aren't in active support anymore.  Even if 5.6 and 7.0 get security fixes, reach their end of life on Dec 2018.
See [php.net/supported-versions](http://php.net/supported-versions.php) for more.

Comment (bzoks) [reply .travis.yml]: I think that since this is Symfony-related, version constraints should be aligned with constraints Symfony enforces. If this version supports Symfony 3.4, it should support same php versions as Symfony 3.4 (5.5+)

Comment (phansys) [reply .travis.yml]: >Its right to drop support for php versions <= 7.0

IMO, it dependens which version this PR will affect. If `nostromo` is meant for `3.x` branch and `master` will alias `4.x`, I agree that we can accept non BC changes like this. If that is the case, I'll be happy to submit a PR for the current latest stable version (`nostromo`) allowing symfony 4 support.

**Comment by andres-montanez** [comment]
Hi all, thanks for the PR, but this will brake a lot of things.
Symfony 4 will be supported in Magallanes 4.
Regards.

**Comment by bzoks** [comment]
@andres-montanez , FYI, this fork works great with Symfony 4: https://github.com/sheeep/Magallanes/tree/features/symfony-4-compatibility
